beacon-chain/state/stategen/getter.go

```diff
@@ -35,12 +35,6 @@ func (s *State) StateByRoot(ctx context.Context, blockRoot [32]byte) (*state.Bea
 	if blockRoot == params.BeaconConfig().ZeroHash {
 		return s.beaconDB.State(ctx, blockRoot)
 	}
-
-	// Short cut if the cachedState is already in the DB.
```

Comment:
Note. This begins to become a limitation because we start saving hot state DB to disk every 128 slots